src/web/app/pages-instructor/instructor-courses-page/add-course-form/add-course-form.component.ts

```diff
@@ -100,4 +109,35 @@ export class AddCourseFormComponent implements OnInit {
   closeEditFormHandler(): void {
     this.closeCourseFormEvent.emit();
   }
+
+  /**
+   * Handles copying from other courses event
+   */
+  onCopy(): void {
+    this.courseService
+      .getAllCoursesAsInstructor('active')
+      .subscribe(async (courses: Courses) => {
+        const modalRef: NgbModalRef = this.ngbModal.open(CopyCourseModalComponent);
+        modalRef.componentInstance.isCopyFromOtherSession = true;
+        modalRef.componentInstance.courses = courses.courses;
+
+        courses.courses.forEach((course: Course) => {
+          this.feedbackSessionsService
+            .getFeedbackSessionsForInstructor(course.courseId)
+            .subscribe((feedbackSessions: FeedbackSessions) => {
+              modalRef.componentInstance.courseToFeedbackSession[course.courseId]
+                = [...feedbackSessions.feedbackSessions];
+            });
+        }, (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorToast(resp.error.message);
+        });
```

Comment:
Looks like it will fan out into many API call at once. I'd keep an eye on it.
What if there are too many sessions to copy? Will it cause new instances to be spun up?
@rrtheonlyone